ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/DbChangelog.hs

```diff
@@ -0,0 +1,430 @@
+{-# LANGUAGE DataKinds                #-}
+{-# LANGUAGE DeriveAnyClass           #-}
+{-# LANGUAGE DeriveGeneric            #-}
+{-# LANGUAGE FlexibleContexts         #-}
+{-# LANGUAGE FlexibleInstances        #-}
+{-# LANGUAGE MultiParamTypeClasses    #-}
+{-# LANGUAGE NamedFieldPuns           #-}
+{-# LANGUAGE ScopedTypeVariables      #-}
+{-# LANGUAGE StandaloneDeriving       #-}
+{-# LANGUAGE StandaloneKindSignatures #-}
+{-# LANGUAGE UndecidableInstances     #-}
+
+module Ouroboros.Consensus.Storage.LedgerDB.HD.DbChangelog (
```

Comment:
I wonder if we should start adding comments before the `module` section. In particular here we could mention briefly what a `DbChangelog` is, on which paper this is based (or Duncan's tech report), and for more info refer the reader to `DbChangelog`.